system/core/base.php

```diff
@@ -328,7 +328,7 @@ function hypha_getUserById($id) {
 		$username - identifier for the user
 	*/
 	function hypha_getUserByName($username) {
-		foreach(hypha_getUserList() as $user) if ($user->getAttribute('username') == $username) return $user;
+		foreach(hypha_getUserList() as $user) if (strtolower($user->getAttribute('username')) == strtolower($username)) return $user;
```

Comment:
This also means that no two users with usernames only differing in case can ever exist. Does user registration use this same function to guarantee this?

Reply:
Yes, no two users with  usernames only differing in case can ever exist.
Yes, the user registration uses this same function to guarantee this.